local/experimental/blockchain.go

```diff
@@ -0,0 +1,490 @@
+				zap.String("subnetID", subnetID.String()),
+				zap.String("txID", txID.String()),
+			)
+		}
+
+		// Wait for validators to become active
+		for ctx.Err() == nil {
+			vs, err := w.client.GetCurrentValidators(ctx, subnetID, nil)
+			if err != nil {
+				return err
+			}
+			curValidators := make(map[ids.NodeID]struct{})
+			for _, v := range vs {
+				curValidators[v.NodeID] = struct{}{}
+			}
+			ready := true
+			for _, nodeID := range nodeIDs {
+				if _, ok := curValidators[nodeID]; !ok {
+					ready = false
+				}
+			}
+			if ready {
+				break
+			}
+			time.Sleep(1 * time.Second)
+			w.log.Info("waiting for all subnet validators to become active")
+		}
+	}
+	return nil
+}
+
+func CreateSpecificBlockchains(
+	ctx context.Context,
+	ln LocalNetwork,
+	execPath string,
+	chainSpecs []network.BlockchainSpec, // VM name + genesis bytes
+	redirectOutput bool,
+) ([]network.BlockchainSpec, []ids.ID, error) {
+	clientURI, err := ln.GetClientURIUnsafe()
+	if err != nil {
+		return nil, nil, err
+	}
+	wallet, err := newWallet(ctx, clientURI, ln.Logger())
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Create new subnets
+	subnets, err := wallet.createSubnets(ctx, len(chainSpecs))
+	if err != nil {
+		return nil, nil, err
+	}
+	for i, subnet := range subnets {
+		subnetIDStr := subnet.String()
+		chainSpecs[i].SubnetID = &subnetIDStr
+	}
+
+	// Ensure all participants exist
+	allNames := map[string]struct{}{}
+	for _, spec := range chainSpecs {
+		for _, name := range spec.Participants {
+			if err := ln.AddNodeUnsafe(node.Config{
+				Name:           name,
+				BinaryPath:     execPath,
+				RedirectStdout: redirectOutput,
+				RedirectStderr: redirectOutput,
+				Flags: map[string]interface{}{
+					config.TrackSubnetsKey: *spec.SubnetID,
+				},
+			}); err != nil {
+				return nil, nil, err
+			}
+			ln.Logger().Info("created node", zap.String("name", name))
+			allNames[name] = struct{}{}
+		}
+	}
+
+	// Ensure all participants are validators
+	if err := wallet.addPrimaryValidators(ctx, ln, allNames); err != nil {
+		return nil, nil, err
+	}
+
+	// Add subnet validators
+	if err := wallet.addSubnetValidators(ctx, ln, chainSpecs); err != nil {
+		return nil, nil, err
+	}
+
+	// Create and store chain configs
+	blockchainTxs, err := wallet.createBlockchainTxs(ctx, chainSpecs)
+	if err != nil {
+		return nil, nil, err
+	}
+	chainIDs := make([]ids.ID, len(blockchainTxs))
+	for i, tx := range blockchainTxs {
+		chainIDs[i] = tx.ID()
+	}
+	ln.SetBlockchainConfigFilesUnsafe(chainSpecs, blockchainTxs)
+
+	// Restart validators
+	if err := restartNodes(ctx, ln, chainSpecs); err != nil {
+		return nil, nil, err
+	}
+	clientURI, err = ln.GetClientURIUnsafe()
+	if err != nil {
+		return nil, nil, err
+	}
+	wallet.reload(clientURI)
+
+	// Create blockchains
+	// if err := ln.ReloadVMPluginsUnsafe(ctx); err != nil {
+	// 	return nil, nil, err
+	// }
+	if err := wallet.createBlockchains(ctx, chainSpecs, blockchainTxs); err != nil {
```

Comment:
it is nice to see the same wallet can be used to issue the txs after it was also used to create those same tx with UTXOs and tx tracking